monitoring/oplog-populator/dashboard.py

```diff
@@ -0,0 +1,335 @@
+    colorMode='value',
+    noValue='0',
+    reduceCalc='last',
+    graphMode='none',
+    targets=[Target(
+        expr='sum(' + Metrics.CONNECTORS.raw(isOld='true') + ')',
+    )],
+    thresholds=[
+        Threshold('red', 0, 0.),
+        Threshold('green', 1, 1.),
+    ])
+
+total_number_buckets = Stat(
+    title='Total Buckets',
+    description='Number of buckets configured on the connectors',
+    dataSource='${DS_PROMETHEUS}',
+    colorMode='value',
+    noValue='0',
+    reduceCalc='last',
+    graphMode='none',
+    targets=[Target(
+        expr='sum(' + Metrics.BUCKETS.raw() + ')',
+    )],
+    thresholds=[
+        Threshold('green', 0, 0.),
+    ])
+
+bucket_distribution = PieChart(
+    title='Bucket Distribution',
+    description='How buckets are distributed among the connectors',
+    dataSource='${DS_PROMETHEUS}',
+    legendValues=['percent', 'value'],
+    pieType='donut',
+    targets=[Target(
+        expr='sum by(connector) (' + Metrics.BUCKETS.raw() + ')',
+        legendFormat='{{connector}}',
+    )])
+
+max_pipeline_size = Stat(
+    title='Max Pipeline Size Per Connector',
+    description='Maximum size of the mongo pipeline per connector',
+    dataSource='${DS_PROMETHEUS}',
+    colorMode='value',
+    noValue='0',
+    reduceCalc='last',
+    format=UNITS.BYTES,
+    orientation='horizontal',
+    graphMode='none',
+    targets=[Target(
+        expr='sum by(connector) (' + Metrics.PIPELINE_SIZE.raw() + ')',
+        legendFormat='{{connector}}',
+    )],
+    thresholds=[
+        Threshold('green', 0, 0.),
+    ])
+
+bucket_additions = TimeSeries(
+    title='Bucket Addition Rate',
+    description='Rate at which buckets are getting added to the connectors',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum by(connector) (rate(' + Metrics.CONNECTOR_CONFIG(opType="add") + '))',
+        legendFormat='{{connector}}',
+    )])
+
+bucket_removals = TimeSeries(
+    title='Bucket Removal Rate',
+    description='Rate at which buckets are getting removed from the connectors',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum by(connector) (rate(' + Metrics.CONNECTOR_CONFIG(opType="delete") + '))',
+        legendFormat='{{connector}}',
+    )])
+
+ack_lag = TimeSeries(
+    title='Acknowledgement Lag By Operation Type',
+    description='Delta between a config change in mongo and it getting processed by the oplogPopulator in seconds',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.SECONDS,
+    targets=[Target(
+        expr='sum by(opType) (rate(' + Metrics.ACK_LAG.sum() + '))',
```

Comment:
I am not sure that value is correct : the _average_ lag (per connector) is `Metrics.ACK_LAG.sum()/Metrics.ACK_LAG.count()`, but this format is the rate at which the cumulated lag increases...

We can display that average value, either as a panel and/or as a graph.
And it may be useful to use the "full" histogram metric to display the repartition of the lag as a heatmap